docs/reference/settings/security-settings.asciidoc

```diff
@@ -1316,6 +1320,32 @@ a PKCS#12 container includes trusted certificate ("anchor") entries look for
 `openssl pkcs12 -info` output, or `trustedCertEntry` in the
 `keytool -list` output.
 
+===== PKCS#11 tokens
+
+When using a PKCS#11 cryptographic token, which contains the
+private key, certificate, and certificates that should be trusted, use
+the following settings:
+
+`xpack.ssl.keystore.type`::
+Set this to `PKCS11`.
+
+`xpack.ssl.truststore.type`::
+Set this to `PKCS11`.
+
+
+[[pkcs11-truststore-note]]
+[NOTE]
+When configuring the PKCS#11 token that your JVM is configured to use as
+a keystore or a truststore for Elasticsearch, the PIN for the token can be
+configured by setting the appropriate value to `xpack.ssl.truststore.password`
+or `xpack.ssl.truststore.secure_password`. In the absence of the above, {es} will
+fallback to use he appropriate JVM setting (`-Djavax.net.ssl.trustStorePassword`)
+if that s set.
+Since there can only be one PKCS#11 token configured, only one keystore and
```

Comment:
So it is a single PKCS11 token per provider. I am fine with the single limitation now and we can always add support for a keystore provider setting so that multiple can be used separately. Given this, maybe the docs should say `Elasticsearch only supports reading the PKCS#11 token from the first PKCS#11 security provider; this means only one keystore ...`

Reply:
https://github.com/elastic/elasticsearch/issues/33459#issuecomment-425496772 explains why it is really only a single one.